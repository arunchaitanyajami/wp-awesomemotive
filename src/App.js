import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import 'react-tabs/style/react-tabs.css';
import { useState } from '@wordpress/element';
import * as qs from 'query-string';

export default () => {
	const { getTabIndex } = qs.parse( location.search );
	const [ tabIndex, setTabIndex ] = useState( getTabIndex ? getTabIndex : 0 );

	const changeTabIndex = ( index ) => {
		setTabIndex( index );

		let newUrl =
			window.location.protocol +
			'//' +
			window.location.host +
			window.location.pathname +
			'?page=awesome-motive&getTabIndex=' +
			index;
		window.history.pushState( { path: newUrl }, '', newUrl );
	};

	return (
		<Tabs
			defaultIndex={ tabIndex }
			onSelect={ ( index ) => changeTabIndex( index ) }
		>
			<TabList>
				<Tab>Settings</Tab>
				<Tab>Table</Tab>
				<Tab>Graph</Tab>
			</TabList>
			<TabPanel>Content 1</TabPanel>
			<TabPanel>Content 2</TabPanel>
			<TabPanel>Content 3</TabPanel>
		</Tabs>
	);
};
