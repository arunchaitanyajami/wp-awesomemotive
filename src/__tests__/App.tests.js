import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import App from '../App'

beforeAll(() => {})
afterEach(() => {})
afterAll(() => {})
describe('App', () => {
  test('Is Expected to have Settings, Graph and Table Tab', async () => {
    let { container } = render(<App/>)
    expect(container.querySelector('#react-tabs-0')).toHaveTextContent('Settings')
    expect(container.querySelector('#react-tabs-2')).toHaveTextContent('Table')
    expect(container.querySelector('#react-tabs-4')).toHaveTextContent('Graph')
    expect(container).toMatchSnapshot()
  })
})
