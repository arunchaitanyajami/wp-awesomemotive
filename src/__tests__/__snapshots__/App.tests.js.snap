// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Is Expected to have Settings, Graph and Table Tab 1`] = `
<div>
  <div
    class="react-tabs"
    data-rttabs="true"
  >
    <ul
      class="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="react-tabs__tab react-tabs__tab--selected"
        data-rttab="true"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        Settings
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-rttab="true"
        id="react-tabs-2"
        role="tab"
      >
        Table
      </li>
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="false"
        class="react-tabs__tab"
        data-rttab="true"
        id="react-tabs-4"
        role="tab"
      >
        Graph
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      Content 1
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
    <div
      aria-labelledby="react-tabs-4"
      class="react-tabs__tab-panel"
      id="react-tabs-5"
      role="tabpanel"
    />
  </div>
</div>
`;
